ledger-core/virtual/integration/method_test.go

```diff
@@ -1534,3 +1535,179 @@ func TestVirtual_Method_ForObjectWithMissingState(t *testing.T) {
+			)
+
+			if test.expectedUnImplementedError {
+				server, ctx = utils.NewUninitializedServerWithErrorFilter(nil, t, func(s string) bool {
+					return !strings.Contains(s, "execution: not implemented")
+				})
+			} else {
+				server, ctx = utils.NewUninitializedServer(nil, t)
+			}
+
+			defer server.Stop()
+
+			executeDone := server.Journal.WaitStopOf(&execute.SMExecute{}, 1)
+
+			runnerMock := logicless.NewServiceMock(ctx, t, nil)
+			server.ReplaceRunner(runnerMock)
+
+			server.Init(ctx)
+			server.IncrementPulse(ctx)
+
+			var (
+				class       = gen.UniqueGlobalRef()
+				objectRef   = server.BuildRandomOutgoingWithPulse()
+				outgoingRef = server.BuildRandomOutgoingWithPulse()
+			)
+
+			dirtyState := test.dirtyStateBuilder(objectRef, class, server.GetPulse().PulseNumber)
+			validatedState := test.validatedStateBuilder(objectRef, class, server.GetPulse().PulseNumber)
+
+			{ // send object state to server
+				pl := payload.VStateReport{
+					Status:               payload.Ready,
+					Object:               objectRef,
+					LatestValidatedState: validatedState.HeadRef(),
+					LatestDirtyState:     dirtyState.HeadRef(),
+					ProvidedContent: &payload.VStateReport_ProvidedContentBody{
+						LatestValidatedState: &payload.ObjectState{
+							Reference: validatedState.StateID(),
+							Class:     class,
+							State:     validatedState.Memory(),
+						},
+						LatestDirtyState: &payload.ObjectState{
+							Reference: dirtyState.StateID(),
+							Class:     class,
+							State:     dirtyState.Memory(),
+						},
+					},
+				}
+
+				server.WaitIdleConveyor()
+				server.SendPayload(ctx, &pl)
+				server.WaitActiveThenIdleConveyor()
+			}
+
+			typedChecker := server.PublisherMock.SetTypedChecker(ctx, mc, server)
+
+			{
+				typedChecker.VCallResult.Set(func(res *payload.VCallResult) bool {
+					require.Equal(t, objectRef, res.Callee)
+					assert.Equal(t, test.callResult, res.ReturnArguments)
+					return false // no resend msg
+				})
+
+				pl := payload.VCallRequest{
+					CallType:            payload.CTMethod,
+					CallFlags:           test.callFlags,
+					Caller:              server.GlobalCaller(),
+					Callee:              objectRef,
+					CallSiteDeclaration: class,
+					CallSiteMethod:      "Test",
+					CallOutgoing:        outgoingRef,
+				}
+
+				key := pl.CallOutgoing.String()
+
+				runnerMock.AddExecutionClassify(key, contract.MethodIsolation{
+					Interference: test.callFlags.GetInterference(),
+					State:        test.callFlags.GetState(),
+				}, nil)
+
+				// if we use forbidden isolation, then execute should stop before Start happen
+				if !test.expectedUnImplementedError {
+					result := requestresult.New(test.callResult, outgoingRef)
+					result.SetAmend(dirtyState, []byte("new stuff"))
+					runnerMock.AddExecutionMock(key).
+						AddStart(nil, &execution.Update{
+							Type:   execution.Done,
+							Result: result,
+						})
+
+				}
+
+				server.SendPayload(ctx, &pl)
+			}
+
```

Comment:
maybe you need to change pulse and check in VStateReport that object state was(not) changed?

Reply:
don't need to. I'm checking that runner was never called.